### PR TITLE
Replace '@' prefix in map name with 'AT_' when creating map

### PR DESCRIPTION
**Pull request by omeranson**
When calling the API function to create the map, e.g., bcc_create_map,
replace the '@' prefix, if it exists, with the string 'AT_'. Otherwise,
map creation with the name fails, and (e.g.,) BCC falls back to
creating the map without a name.

Closes: #1752

##### Checklist

- [ ] Language changes are updated in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -13,11 +13,15 @@
...
+  std::string name_ = name;
+  if (name_[0] == '@') {
+    name_ = "AT_" + name.substr(1);
+  }
 #ifdef HAVE_BCC_CREATE_MAP
-  return bcc_create_map(map_type, name, key_size, value_size, max_entries, flags);
+  return bcc_create_map(map_type, name_.c_str(), key_size, value_size, max_entries, flags);
 #else
-  return bpf_create_map(map_type, name, key_size, value_size, max_entries, flags);
+  return bpf_create_map_name(map_type, name_.c_str(), key_size, value_size, max_entries, flags);
```
Comment (danobi) [inline src/map.cpp:24]: The `#ifdef HAVE_BCC_CREATE_MAP` is to handle when bcc renamed its own `bpf_create_map` -> `bcc_create_map`. Changing this to libbpf's `bpf_create_map_name` may have unintended consequences when bpftrace is compiled without libbpf.

All that to say: please undo this change :)

File: src/map.cpp [hunk 2]
```diff
@@ -13,11 +13,15 @@
 
 namespace bpftrace {
 
-int Map::create_map(enum bpf_map_type map_type, const char *name, int key_size, int value_size, int max_entries, int flags) {
+int Map::create_map(enum bpf_map_type map_type, const std::string & name, int key_size, int value_size, int max_entries, int flags) {
+  std::string name_ = name;
```
Comment (danobi) [inline src/map.cpp:17]: Style nit:
```suggestion
  std::string fixed_name = name;
```
We generally use trailing `_` for member variables, not locals.

File: src/map.cpp [hunk 3]
```diff
@@ -13,11 +13,15 @@
...
+  if (name_[0] == '@') {
```
Comment (danobi) [inline src/map.cpp:18]: Should check `!fixed_name.empty()` before indexing

**Review by danobi — CHANGES_REQUESTED** [review]
Thanks for the fix; few comments.

Also please address the clang-format lint -- our policy is to re-format old code when the old code is modified.

File: src/map.cpp (hunk 1 above)
Comment (omeranson) [reply src/map.cpp]: Done.

File: src/map.cpp (hunk 2 above)
Comment (omeranson) [reply src/map.cpp]: Done

File: src/map.cpp (hunk 3 above)
Comment (omeranson) [reply src/map.cpp]: Done. Although C++11 allows indexing the one-passed-the-end character of a string.

**Comment by omeranson** [comment]
Thanks for the review. I went ahead and passed map.cpp via clang-format, since I didn't want to guess the exact formatting.

**Review by danobi — APPROVED** [review]
Thank you!
